test/unit/mriController.test.js

```diff
@@ -0,0 +1,343 @@
+                page: 1
+            },
+            user: {
+                username: ''
+            },
+            isAuthenticated: function() {
+                return this.user.username ? true : false;
+            },
+            isTokenAuthenticated: false
+        };
+        let resSpy = sinon.spy();
+        let jsonSpy = sinon.spy();
+        let res = {
+            send: resSpy,
+            status: sinon.stub().returns({ json: jsonSpy}),
+            json: jsonSpy
+        };
+        await mriController.apiMriGet(req, res);
+        assert.notDeepStrictEqual(resSpy.calledOnce, true);
+        assert.notDeepStrictEqual(jsonSpy.calledOnce, true);
+        sinon.restore();
+    });
+
+    it('apiMriGet function() should throw an error when the URL is invalid', async function() {
+        let req = {
+            db: db,
+            query: {
+                url: '',
+                download: 'true',
+                backups: 'true',
+                page: 1
+            },
+            user: {
+                username: ''
+            },
+            isAuthenticated: function() {
+                return this.user.username ? true : false;
+            },
+            isTokenAuthenticated: false
+        };
+        let resSpy = sinon.spy();
+        let jsonSpy = sinon.spy();
+        let res = {
+            send: resSpy,
+            status: sinon.stub().returns({ json: jsonSpy}),
+            json: jsonSpy
+        };
+        await mriController.apiMriGet(req, res);
+        assert.notDeepStrictEqual(resSpy.calledOnce, true);
+        assert.notDeepStrictEqual(jsonSpy.calledOnce, true);
+        sinon.restore();
+    });
+
+    it('apiMriPost function() should work correctly and make the right calls when input is correct', async function() {
+        let req = {
+            db: db,
+            body: {},
+            query: {
+                url: 'https://brainbox.pasteur.fr/mri?url=https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz',
+            },
+            user: {
+                username: ''
+            },
+            dirname: __dirname.split('/test')[0],
+            isAuthenticated: function() {
+                return this.user.username ? true : false;
+            },
+            isTokenAuthenticated: false
+        };
+        let resSpy = sinon.spy();
+        let jsonSpy = sinon.spy();
+        let res = {
+            send: resSpy,
+            status: sinon.stub().returns({ json: jsonSpy}),
+            json: jsonSpy
+        };
+        await mriController.apiMriPost(req, res);
+        assert.notDeepStrictEqual(resSpy.calledOnce, true);
+        assert.notDeepStrictEqual(jsonSpy.calledOnce, true);
+        sinon.restore();
+    });
+
+    it('apiMriPost function() should throw an error when input is incorrect', async function() {
+        let req = {
+            db: db,
+            body: {},
+            query: {
+                url: '',
```

Comment:
a POST to url: '' will return a paginated list of available MRIs (in this case, page=1), and not an error. The test should be split in two: One for testing that paginated MRI lists are working, and one for testing that an incorrect URL does fail. Use an invalid URL, like `url: "invalid_url"`.